sklearn/metrics/tests/test_pairwise.py

```diff
@@ -110,6 +111,19 @@ def test_pairwise_distances(global_dtype):
     S2 = haversine_distances(X, Y)
     assert_allclose(S, S2)
 
+    # test paired haversine
+    X = rng.randn(5, 3)
+    Y = rng.randn(5, 3)
+    with pytest.raises(ValueError):
+        paired_haversine_distances(X, Y)
+
+    # compare paired to pairwise implementations
+    X = rng.randn(5, 2)
+    Y = np.tile(X[0, :], (5, 1))
+    S = haversine_distances(X, Y[0].reshape(1, -1)).ravel()
+    S2 = paired_haversine_distances(X, Y).ravel()
+    assert_allclose(S, S2)
```

Comment:
For consistency, do you mind moving the `paired_haversine` tests together with the other paired tests?
https://github.com/scikit-learn/scikit-learn/blob/348f01ab3f10852906adc168460e4c342a4ae471/sklearn/metrics/tests/test_pairwise.py#L1184

Reply:
Hm but there are more paired distance tests right after these lines?

Reply:
Sorry I don't understand your answer ... so probably my comment wasn't clear.
What I meant is that tests on `paired_<some metric>` functions are done after line 1184 (now 1018) , I am suggesting to move the test on the `paired_haversine` there... does that make sense?

Reply:
Right, I think I get it.